e3/job/__init__.py

```diff
@@ -107,8 +144,12 @@ def run(self):
         # Do non blocking spawn followed by a wait in order to have
         # self.proc_handle set. This allows support for interrupt.
         cmd_options['bg'] = True
-        self.proc_handle = Run(self.cmdline, **cmd_options)
-        self.proc_handle.wait()
+        with self._lock:
```

Comment:
You cannot avoid the check on self.interrupted here as the following scenario can occurs:
A- (lock) check for interrupted
B- (lock) set to interrupted
C- (lock) run